src/blog/2024/04/integrate-dynamodb-with-node-red.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Integrate Node-RED with DynamoDB
+subtitle: Get started with DynamoDB with Node-RED
+description: Get started with DynamoDB with Node-RED
+date: 2024-04-17
+authors: ["grey-dziuba"]
+# image: /blog/2024/04/images/flowfuse-at-hannover-messe-node-red.png
+tags:
+    - posts
+    - flowfuse
+    - node-red
+    - dynamodb
+    - nosql
+    - aws
+---
+
+Amazon's DynamoDB is a fully managed NoSQL database service known for its fast and predictable performance and scalable design. This makes it suitable for applications needing consistent, low-latency responses. In this blog, we’ll look at how to set up and use DynamoDB, configure the necessary IAM roles, and apply Node-RED flows to store and retrieve data effectively.
+
+
+<!--more-->
+
+## Setting Up DynamoDB
+### Creating a DynamoDB Table
+
+To begin with DynamoDB, you first need to create a table. Here’s how you can do it:
+
+1. Log into the AWS Management Console and navigate to the DynamoDB section.
+2. Click on **Create table**.
+3. For the table name, enter a table name. We will use **FlowFuse-Email** for this demonstration. For the partition key, use **email** with type String.
+4. We will keep the **default settings**.
+5. Finally, click on **Create** to establish your table.
+
+
+### Configuring IAM for DynamoDB
+
+Proper permissions are critical for secure and efficient operation:
```

Comment:
Can we also explain what IAM is, why you need to set it up? When I create a SQLite DB in Node-RED I don't need any of this, so it might be a good one for the user to understand why they have all these extra steps now.